src/infra/core/tests/tests/ingest/test_polling_ingest.rs

```diff
@@ -954,6 +869,188 @@ async fn test_ingest_polling_datafusion_bad_column_names_rename() {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+#[test_group::group(containerized, engine, ingest, spark)]
+#[test_log::test(tokio::test)]
+async fn test_ingest_polling_preprocess_with_spark() {
```

Comment:
Large number of duplicated actions between Spark and Flink test.

Reply:
yeah, I tried generalizing the test, but since different engines can have slightly different queries and produce slightly different schemas - most of parameters end up to be non-reusable and the amount of lines of code saved is not that big.